topology/src/main/java/io/camunda/zeebe/topology/TopologyInitializer.java

```diff
@@ -96,20 +134,17 @@ public FileInitializer(final Path topologyFile, final ClusterTopologySerializer
     @Override
     public ActorFuture<ClusterTopology> initialize() {
       try {
-        if (Files.exists(topologyFile)) {
-          final var serializedTopology = Files.readAllBytes(topologyFile);
-          if (serializedTopology.length > 0) {
-            final var clusterTopology = serializer.decodeClusterTopology(serializedTopology);
-            LOGGER.debug(
-                "Initialized cluster topology '{}' from file '{}'", clusterTopology, topologyFile);
-            return CompletableActorFuture.completed(clusterTopology);
-          }
+        final var persistedTopology =
+            PersistedClusterTopology.ofFile(topologyFile, serializer).getTopology();
+        if (!persistedTopology.isUninitialized()) {
+          LOGGER.debug(
+              "Initialized cluster topology '{}' from file '{}'", persistedTopology, topologyFile);
         }
+        return CompletableActorFuture.completed(persistedTopology);
       } catch (final Exception e) {
-        return CompletableActorFuture.completedExceptionally(e);
+        return CompletableActorFuture.completedExceptionally(
+            new PersistedTopologyIsBroken(topologyFile, e));
```

Comment:
:question: If we're falling back to the `SyncInitializer` on any `PersistedTopologyIsBroken`, but we convert all exceptions here to that anyway, why the fallback only on some exception class? :thinking:

Reply:
Yeah, perhaps a _bit_ over-engineered but I think I still like it this way because it communicates the intent a bit better and prevents us from accidentally using the recovery via sync for new exception types when we change the `FileInitializer`